extension/test/unit/lib/login-test.js

```diff
@@ -0,0 +1,97 @@
+const chai = require('chai')
+const sinon = require('sinon')
+const proxyquire = require('proxyquire')
+
+// Pre-loading the module with all its dependencies in order to make the proxyquire call below
+// over a hundred times faster, so that it doesn't cause a timeout.
+require('../../../lib/login')
+
+const BigCommerceCustomerRepository = require('../../../lib/bigcommerce/CustomerRepository')
+
+describe('login()', async () => {
+  const sandbox = sinon.createSandbox()
+  let login
+  let requestStub
+  let repoStub
+
+  const context = {
+    config: {
+      clientId: '',
+      accessToken: '',
+      storeHash: ''
+    }
+  }
+  const input = {
+    shopUrl: 'https://store-r5s844ad.mybigcommerce.com',
+    parameters: {
+      login: 'john.doe@test.com',
+      password: 'password1'
+    }
+  }
+
+  beforeEach(function () {
+    requestStub = sandbox.stub()
+    login = proxyquire('../../../lib/login', {
+      'request-promise-native': requestStub
+    })
+
+    repoStub = sandbox.createStubInstance(BigCommerceCustomerRepository)
+    sandbox.stub(BigCommerceCustomerRepository, 'create').returns(repoStub)
+  })
+
+  afterEach(() => {
+    sandbox.verifyAndRestore()
+  })
+
+  it('should return user data', async () => {
```

Comment:
I think we should cover more cases here. I've just changed the code within and had no way to check against tests if the flow is still correct.

Reply:
done